### PR TITLE
feat(search)

### PR DESCRIPTION
**Pull request by ajbt200128**
Added a search tab + functionality. Also, figured out how to gate features and commands by semgrep CLI version!

Sister PR [here](https://github.com/returntocorp/semgrep/pull/7938)
### Security

- [ ] Change has security implications (if so, ping the security team)

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -7,3 +7,43 @@ export const LSP_LOG_FILE = "semgrep_lsp.log";
 export const VSCODE_CONFIG_KEY = "semgrep";
 export const VSCODE_EXT_NAME = CLIENT_NAME;
 export const MIN_VERSION = ">=1.21.0"; // Minimum semgrep version required for this extension
+export const LATEST_VERSION = ">=1.25.0";
```
Comment (tpetr) [inline src/constants.ts:10]: can we make this variable a little more descriptive? (latest version for what?)

File: src/constants.ts [hunk 2]
```diff
@@ -7,3 +7,43 @@ export const LSP_LOG_FILE = "semgrep_lsp.log";
...
+export const SUPPORTED_LANGS = [
```
Comment (tpetr) [inline src/constants.ts:11]: short-term this is fine, but we should think about if we can get this data from semgrep directly so there's one source of truth)

File: src/ruleExplorer.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import {
+/*import {
```
Comment (tpetr) [inline src/ruleExplorer.ts:1]: can we delete instead of commenting out?

**Review by tpetr — APPROVED** [review]
just some nits, LGTM

File: src/constants.ts (hunk 2 above)
Comment (ajbt200128) [reply src/constants.ts:11]: Agree, there's just no direct way rn, but I'll probably fix in the future if we use this for anything besides search

Comment (aryx) [reply src/constants.ts:11]: lang.json in the semgrep-interfaces repo has this information (you can add it as a submodule to this repo?)

**Pull request merged**